feat: 기본 배송지 저장 기능 추가, JWT 발급 기능 추가

Pascal-Case opened the pull request:
### 변경사항

**AS-IS**
- 회원 가입시 주소를 회원 엔티티에만 저장하도록 되어있었음
- 로그인 기능 미구현 

**TO-BE**
- 회원 가입시 입력하는 주소로 기본 배송지를 추가하도록 기능 추가
- 입력한 주소를 카카오 local api를 이용하여 좌표로 변환하여 저장
- 고객, 점주, 라이더에 대해 각각 다른 로그인 필터를 적용하기 위해 보안 설정을 분리. 일부 컨트롤러 설계도 변경
- 로그인 성공 시 JWT 발급 기능 추가. 인증 여부 테스트를 위한 컨트롤러 추가

### 테스트

- [x] 테스트 코드
- [x] API 테스트 

#3

Pascal-Case committed:
feat: 유저 생성 시 기본 배송지 저장 기능 추가

Pascal-Case committed:
test: 기본 배송지 저장 기능 테스트 코드 작성

Pascal-Case committed:
refactoring: 로그 코드 추가

Pascal-Case committed:
refactoring: 로그 코드 추가

Pascal-Case committed:
feat: JWT 토큰 적용

- User, Partner, Rider 별 보안 설정 분리
- 커스텀 로그인 필터, 인증, 토큰 발급 기능 추가
- 일부 구조 변경 및 불필요한 코드 제거

Pascal-Case committed:
feat: cors 설정 추가

Pascal-Case committed:
test: JWT 토큰 발급 관련 테스트 코드 추가

- JWT 토큰 발급 테스트 코드 추가
- 기존 테스트 코드 일부 수정

minsoo0506 commented:
벌써 swagger 설정과 logger 적용까지 완료하셨네요!! test 코드까지 작성하셨는데 고생 많으셨습니다! 항상 제가 미루게 되는 3가지인데 저도 이번주 안까지 코드에 적용 시켜보도록 하겠습니다!

Pascal-Case commented:
> 벌써 swagger 설정과 logger 적용까지 완료하셨네요!! test 코드까지 작성하셨는데 고생 많으셨습니다! 항상 제가 미루게 되는 3가지인데 저도 이번주 안까지 코드에 적용 시켜보도록 하겠습니다!

저도 항상 미뤄왔었는데 나중엔 감당이 안돼서 이번엔 프로젝트 초기에 도입해 봤습니다. 화이팅 !

Pascal-Case committed:
fix: cors allowedOrigins 오류 발생 부분 수정

추후 수정 필요

mingming-mentor commented on src/main/java/jyang/deliverydotdot/dto/location/KakaoGeoResponse.java:
```diff
@@ -0,0 +1,32 @@
+  @AllArgsConstructor
+  public static class Document {
+
+    private String address_name;
```
java convention에 맞춰 카멜케이스로 작성해주시면 좋을 것 같아요

mingming-mentor replied:
만약 kakao쪽 response로 인해 이렇게 설정하신 거라면 @JsonProperty를 사용하시면 좋을 것 같습니다

Pascal-Case replied:
JsonProperty어노테이션을 사용하여 수정하였습니다!

mingming-mentor commented on src/main/java/jyang/deliverydotdot/exception/GlobalExceptionHandler.java:
```diff
@@ -1,25 +1,38 @@
 @RestControllerAdvice
 public class GlobalExceptionHandler {
 
+  private final Logger logger = LoggerFactory.getLogger(GlobalExceptionHandler.class);
```
lombok의 @Slf4j를 이용하셔도 됩니다

mingming-mentor commented on src/main/java/jyang/deliverydotdot/service/CommonUserDetailsService.java:
```diff
@@ -0,0 +1,52 @@
+      case ROLE_USER -> userRepository.findByLoginId(loginId)
+          .map(CommonUserDetails::fromUser)
+          .orElseGet(() -> {
+            logger.error("User not found for ID: {}", loginId);
```
UsernameNotFoundException 를 던지게 되면 GlobalExceptionHandler에서도 error를 한번 찍게 되는 것으로 이해하고 있어요. 그렇다면 이쪽 로직에서도 error를 찍어줘야할 필요가 있는지 궁금합니다! 
필요가 있다면 두어도 괜찮지만, 만약 중복된 에러를 찍게 되는 것이라면 한곳에서만 찍도록해도 좋을 것 같아요

Pascal-Case replied:
생각해보니 중복된 에러인거 같습니다. 중복된 에러 제거하고 로거도 slf4j 어노테이션을 붙이는 방식으로 수정했습니다!

mingming-mentor reviewed (approved):
전체적으로 로직을 깔끔하게 잘 작성해주셨네요!! 수고 많으셨습니다

ykchoi1203 commented:
저도 주소 좌표 값을 얻어오려고 카카오 local api랑 google map 중에 고민을 하고 있었는데 카카오로 하셨군요. 고생하셨습니다! 저도 얼른 적용시켜 봐야겠네요. 남은 프로젝트 기간도 화이팅입니다!

Pascal-Case committed:
fix: KakaoGeoResponse address name 수정

- 멘토님 피드백 사항 수정
- java convention에 맞게 카멜케이스로 수정
- kakao geo response 맵핑으로 위해 JsonProperty 어노테이션 추가

Pascal-Case committed:
fix: 로그 관련 피드백 사항 수정

- 멘토님 피드백 사항 수정
- 중복 에러 제거
- @slf4j 어노테이션 추가
